near-plugins/tests/contracts/access_controllable/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+use near_plugins::{access_control, AccessControlRole, AccessControllable};
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::{env, log, near_bindgen, AccountId};
+use std::collections::HashMap;
+use std::convert::TryFrom;
+
+#[derive(AccessControlRole, BorshSerialize, BorshDeserialize)]
+pub enum Role {
+    Level1,
+    Level2,
+    Level3,
+}
+
+#[access_control(storage_prefix = "__foo", role_type = "Role")]
```

Comment:
Would this work if you replace `role_type = "Role"` with `Role`

```suggestion
#[access_control(Role)]
```

Omitted `storage_prefix` for clarity. I understand you don't have access to the trait per-se, but you only need access to the name of the trait.

----

With regards to the `storage_prefix` let's use some sensible default like: `__ACL` (+ allow overriding)

Reply:
> Would this work if you replace role_type = "Role" with Role

I'll try if this is possible with `darling` (which handles argument parsing).

> With regards to the storage_prefix let's use some sensible default like: __ACL (+ allow overriding)

That's implemented :)

```rust
// Can't find a way to insert a link to a new file, so just pasting the code from
// near-plugins-derive/src/access_controllable.rs L49-51
let storage_prefix = macro_args
    .storage_prefix
    .unwrap_or_else(|| DEFAULT_STORAGE_PREFIX.to_string());
```

Changes in e8dfbac:
- Omit `storage_prefix`. Let's only specify _required_ arguments in this contract?
- Then we could add another (very simple) contract in which _all optional_ arguments are set

Reply:
Let's keep it simple for now, as it is after commit LGTM.